core/optaplanner-core-impl/src/main/java/org/optaplanner/core/impl/domain/variable/custom/FooShadowVariableDescriptor.java

```diff
@@ -0,0 +1,136 @@
+package org.optaplanner.core.impl.domain.variable.custom;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.optaplanner.core.api.domain.solution.PlanningSolution;
+import org.optaplanner.core.api.domain.variable.AbstractVariableListener;
+import org.optaplanner.core.api.domain.variable.ShadowVariable;
+import org.optaplanner.core.api.domain.variable.VariableListener;
+import org.optaplanner.core.config.util.ConfigUtils;
+import org.optaplanner.core.impl.domain.common.accessor.MemberAccessor;
+import org.optaplanner.core.impl.domain.entity.descriptor.EntityDescriptor;
+import org.optaplanner.core.impl.domain.policy.DescriptorPolicy;
+import org.optaplanner.core.impl.domain.variable.ListVariableListener;
+import org.optaplanner.core.impl.domain.variable.descriptor.ShadowVariableDescriptor;
+import org.optaplanner.core.impl.domain.variable.descriptor.VariableDescriptor;
+import org.optaplanner.core.impl.domain.variable.listener.VariableListenerWithSources;
+import org.optaplanner.core.impl.domain.variable.supply.Demand;
+import org.optaplanner.core.impl.domain.variable.supply.SupplyManager;
+
+/**
+ * @param <Solution_> the solution type, the class with the {@link PlanningSolution} annotation
+ */
+public class FooShadowVariableDescriptor<Solution_> extends ShadowVariableDescriptor<Solution_> {
```

Comment:
TODO rename this. There is already a `ShadowVariableDescriptor`. Proposal A:
- `ShadowVariableDescriptor` → `AbstractShadowVariableDescriptor`
- `FooShadowVariableDescriptor` → `ShadowVariableDescriptor`

Reply:
Proposal B:
- `ShadowVariableDescriptor` unchanged.
- `FooShadowVariableDescriptor` → `UserShadowVariableDescriptor`.

Breaks the convention where the descriptor class name is the annotation class name `+ Descriptor` suffix. Rename `@ShadowVariable` to `@UserShadowVariable`?

Reply:
Proposal C:

- ShadowVariableDescriptor -> LegacyShadowVariableDescriptor
- FooShadowVariableDescriptor → ShadowVariableDescriptor

Reply:
Proposal D:
like C, but DeprecatedCustomShadowVariableDescriptor

Reply:
Let's discuss at the meeting tomorrow